### PR TITLE
update FileCheck to run on MLIR tests;

### PR DESCRIPTION
**Pull request by ai-mannamalai**
update FileCheck to run on MLIR tests;
Ref: https://discord.com/channels/901152454077452399/1226892677887426560/1271271464493908140

### CODE REVIEW

**Review by asraa — APPROVED** [review]

**Review by asraa — COMMENTED** [review]
It looks like some of those tests caught incorrect tests. Today EOD I can help send a patch to fix!

**Comment by AlexanderViand-Intel** [comment]
> It looks like some of those tests caught incorrect tests. Today EOD I can help send a patch to fix!

For `cggi/syntax.mlir` and `cggi/verifier.mlir`, they simply fail because there are no `//CHECK` commands, but that was intentional for these tests, as they just test successful parsing. Should we add some `//CHECK` to those kind of test cases anyway?

**Comment by asraa** [comment]
> Should we add some //CHECK to those kind of test cases anyway?

A `// CHECK-LABEL` on the function name would be simple for those!

**Comment by ai-mannamalai** [comment]
@asraa @AlexanderViand-Intel - my update fixes few more MLIR file check and leaves only two files in need of update:

```
********************
Failed Tests (2):
  heir :: cggi/straight_line_vectorizer.mlir
  heir :: cggi/verifier.mlir


Testing Time: 0.32s
  Passed: 4
  Failed: 2
```

File: tests/cggi/verifier.mlir [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// RUN: heir-opt --split-input-file --verify-diagnostics %s
+// RUN: heir-opt --split-input-file --verify-diagnostics %s | FileCheck %s
```
Comment (asraa) [inline tests/cggi/verifier.mlir:1]: Actually this one doesn't need the `| FileCheck %s` - this one is using the `--verify-diagnostics` flag, and so it's not a FileCheck test, but asserting on error messages using MLIR's diagnostic tests: https://mlir.llvm.org/getting_started/TestingGuide/#diagnostic-tests

That should fix this one! (As an experiment you can tweak some of the error messages and see that it'll fail)

File: tests/cggi/straight_line_vectorizer.mlir [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // TODO(#519): disable FileChecks until nondeterminism issues are resolved
-// RUN: heir-opt --straight-line-vectorize %s
+// RUN: heir-opt --straight-line-vectorize %s | FileCheck %s
```
Comment (asraa) [inline tests/cggi/straight_line_vectorizer.mlir:2]: For this test: use this gist: https://gist.github.com/asraa/81aaa9dea83a3e67620420c4439b864a

There are some non-determinism issues (linked as a TODO in this test above), but there are only two cases for it, and we can at least check that the `cggi.lut3` was vectorized there.

**Review by asraa — COMMENTED** [review]
Thank you! See fixes

File: tests/cggi/verifier.mlir (hunk 1 above)
Comment (ai-mannamalai) [reply tests/cggi/verifier.mlir:1]: works - basically status quo

**Review by asraa — APPROVED** [review]

**Pull request merged**